blocks/procedures.js

```diff
@@ -1171,14 +1172,15 @@ blocks['procedures_ifreturn'] = {
         this.hasReturnValue_ = true;
       }
       this.setWarningText(null);
-      if (!this.isInFlyout) {
-        this.setEnabled(true);
-      }
     } else {
       this.setWarningText(Msg['PROCEDURES_IFRETURN_WARNING']);
-      if (!this.isInFlyout && !this.getInheritedDisabled()) {
-        this.setEnabled(false);
-      }
+    }
+    if (!this.isInFlyout) {
```

Comment:
Just curious why don't we need the `this.getInheritedDisabled` anymore?

Reply:
I don't think we ever needed it. That would mean that if you dragged an if-return out of a procedure block, and into a disabled block, it wouldn't get disabled itself. Then you could re-enable the disabled parent, and the if-return would get enabled as well, which is not the behavior we want.